include/experimental/__p1673_bits/blas2_matrix_vector_product.hpp

```diff
@@ -350,9 +364,9 @@ void triangular_matrix_vector_product(in_matrix_t A,
     }
   }
   else {
-    for (ptrdiff_t j = 0; j < A.extent(1); ++j) {
-      const ptrdiff_t i_upper = explicitDiagonal ? j : j - ptrdiff_t(1);
-      for (ptrdiff_t i = 0; i <= i_upper; ++i) {
+    for (size_type j = 0; j < A.extent(1); ++j) {
+      const size_type i_upper = explicitDiagonal ? j : j - size_type(1);
```

Comment:
This will break on the first iteration, where `i_upper` will be `size_type(-1)`, i.e., `std::numeric_limits<size_type>::max()`.